services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,21 +46,56 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  BaseType_t checkQueueSend;
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+  checkQueueSend = xQueueSend(thermalMgrQueueHandle, (void *) event, (TickType_t) 10);
+  if (checkQueueSend == pdFAIL) {
+    return ERR_CODE_INVALID_ARG;
+  }
   return ERR_CODE_SUCCESS;
+  
 }
 
 void osHandlerLM75BD(void) {
+  
   /* Implement this function */
+  thermal_mgr_event_t interrupt;
+  interrupt.type = THERMAL_MGR_EVENT_TYPE_INTERRUPT;
+  thermalMgrSendEvent(&interrupt); 
+  
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  float tempData;
+  error_code_t readTempCheck;
+  thermal_mgr_event_t eventTypeCheck; 
```

Comment:
In general try to limit the scope of your variables to be as small as possible, so since these are not used outside the while loop they should not be declared outside of it